authentication/authentication_provider.go

```diff
@@ -1,11 +1,12 @@
 package authentication
 
 import (
+	"context"
 	abs "github.com/microsoft/kiota-abstractions-go"
 )
 
 // AuthenticationProvider authenticates the RequestInformation request.
 type AuthenticationProvider interface {
 	// AuthenticateRequest authenticates the provided RequestInformation.
-	AuthenticateRequest(request *abs.RequestInformation, additionalAuthenticationContext map[string]interface{}) error
```

Comment:
I believe the access token provider will need that parameter as well. And the BearerTokenAuthenticationProvider as well as the AnonymousAuthenticationProvider probably need to be updated.

Reply:
base bearer token authentication provider and access token provider still need to be updated